pkg/securitypolicy/securitypolicyenforcer.go

```diff
@@ -447,6 +470,103 @@ func possibleIndexesForID(containerID string, mapping map[int]map[string]struct{
 	return possibles
 }
 
+// EnforceExpectedMountsPolicy for StandardSecurityPolicyEnforcer injects a
+// hooks.CreateRuntime hook into container spec and the hook ensures that
+// the expected mounts appear prior container start. At the moment enforcement
+// is expected to take place inside LCOW UVM.
+//
+// Supported scenarios:
+//   1. expected mount is provided as a path inside the sandbox, and it should resolve inside UVM
+//      e.g, "sandbox://path/on/the/host", which will correspond to "/run/gcs/c/<podID>/sandboxMounts/path/on/the/host"
+//   2. expected mount is provided as a path under a sandbox mount path inside container, e.g.,
+//      sandbox mount is at path "/sandbox/mount" and wait path is "/sandbox/mount/wait/path", which
+//      corresponds to "/run/gcs/c/<podID>/sandboxMounts/path/on/the/host/wait/path"
+//   3. expected mount is provided as an arbitrary path inside container, and it should resolve
+//      inside UVM, e.g., "/arbitrary/container/path", which corresponds to
+//      "/run/gcs/c/<containerID>/rootfs/arbitrary/container/path"
```

Comment:
I guess I'm confused why we need to support scenarios 1 AND 2. Why can't we just support scenario 1 and require that the user use the `sandboxMounts://` prefix in the expected mounts definition since that's what they'd be adding to the container config so it would be consistent? It feels like to me, supporting both scenarios limits as resolving to sandboxMounts limits our ability to expand this list of supported scenarios going forward.

Reply:
I'm not sure I agree that 3 is an "edge case". It seems like an entirely different scenario to cover. We should drop scenario 1 as mentioned. Should we add 3 in this PR or is that later work?

Reply:
I just left support for scenario 2 and dropped 1, 3. Also added CRI tests.